### PR TITLE
azure: add two presubmit jobs to test changes in test/utils/image/manifest.go

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

Follow-up PR for https://github.com/kubernetes-sigs/windows-testing/pull/249. All Windows presbumit should use https://raw.githubusercontent.com/kubernetes-sigs/windows-testing/master/images/image-repo-list-pr as the image-repo-list.

/assign @jsturtevant

### CODE REVIEW

**Comment by jsturtevant** [comment]
Do we really want all our pre-submits to use the default images?  That would currently block all pr that we wish to run these against.

**Comment by chewong** [comment]
@jsturtevant added two new presubmit jobs - `pull-kubernetes-e2e-aks-engine-azure-windows-test-images` and `pull-kubernetes-e2e-aks-engine-windows-contianerd-test-images` that will be auto-triggered if any changes are introduced to `test/utils/image/manifest.go`.

File: config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml [hunk 1]
```diff
@@ -134,6 +139,116 @@ presubmits:
...
+        - --aksengine-deploy-custom-k8s
+        - --aksengine-agentpoolcount=2
+        # Specific test args
+        - --test_args=--node-os-distro=windows --ginkgo.focus=\[Conformance\]|\[NodeConformance\]|\[sig-windows\]|\[sig-apps\].CronJob|\[sig-api-machinery\].ResourceQuota|\[sig-scheduling\].SchedulerPreemption|\[sig-network\].EndpointSlice.should.create.and.delete.Endpoints.and.EndpointSlices.for.a.Service.with.a.selector.specified|\[sig-network\].EndpointSlice.should.create.Endpoints.and.EndpointSlices.for.Pods.matching.a.Service|\[sig-network\].EndpointSlice.should.have.Endpoints.and.EndpointSlices.pointing.to.API.Server --ginkgo.skip=\[LinuxOnly\]|\[Serial\]|\[Slow\]|GMSA|Guestbook.application.should.create.and.stop.a.working.application|device.plugin.for.Windows
+        - --ginkgo-parallel=4
+        securityContext:
+          privileged: true
+    annotations:
+      testgrid-dashboards: sig-windows-presubmit, provider-azure-windows, provider-azure-upstream, provider-azure-presubmit
+      testgrid-tab-name: pr-aks-engine-azure-windows
```
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml:194]: This name will over lap with the existing pre-submit

File: config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml [hunk 2]
```diff
@@ -134,6 +139,116 @@ presubmits:
...
+        - --aksengine-deploy-custom-k8s
+        - --aksengine-agentpoolcount=2
+        # Specific test args
+        - --test_args=--node-os-distro=windows --ginkgo.focus=\[Conformance\]|\[NodeConformance\]|\[sig-windows\]|\[sig-apps\].CronJob|\[sig-api-machinery\].ResourceQuota|\[sig-scheduling\].SchedulerPreemption|\[sig-network\].EndpointSlice.should.create.and.delete.Endpoints.and.EndpointSlices.for.a.Service.with.a.selector.specified|\[sig-network\].EndpointSlice.should.create.Endpoints.and.EndpointSlices.for.Pods.matching.a.Service|\[sig-network\].EndpointSlice.should.have.Endpoints.and.EndpointSlices.pointing.to.API.Server --ginkgo.skip=\[LinuxOnly\]|\[Serial\]|\[Slow\]|GMSA|Guestbook.application.should.create.and.stop.a.working.application|device.plugin.for.Windows
+        - --ginkgo-parallel=4
+        securityContext:
+          privileged: true
+    annotations:
+      testgrid-dashboards: sig-windows-presubmit, provider-azure-windows, provider-azure-upstream, provider-azure-presubmit
+      testgrid-tab-name: pr-aks-engine-azure-windows-containerd
```
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml:249]: This name will over lap with the existing pre-submit

**Review by jsturtevant — COMMENTED** [review]
This will enable us to test the image before bringing them into our rotation.  Though I think we should have just enable the containerd job. I am worried about to many job configurations as we already have quite a few.

File: config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml [hunk 3]
```diff
@@ -134,6 +139,116 @@ presubmits:
...
+        - --aksengine-creds=$(AZURE_CREDENTIALS)
+        - --aksengine-download-url=https://aka.ms/aks-engine/aks-engine-k8s-e2e.tar.gz
+        - --aksengine-public-key=$(K8S_SSH_PUBLIC_KEY_PATH)
+        - --aksengine-private-key=$(K8S_SSH_PRIVATE_KEY_PATH)
+        - --aksengine-winZipBuildScript=$(WIN_BUILD)
+        - --aksengine-orchestratorRelease=1.20
+        - --aksengine-template-url=https://raw.githubusercontent.com/kubernetes-sigs/windows-testing/master/job-templates/kubernetes_release_staging.json
+        - --aksengine-win-binaries
+        - --aksengine-deploy-custom-k8s
+        - --aksengine-agentpoolcount=2
```
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml:186]: I see that we copied the other pre-submitts but this flag is not in our periodics. Do you know if its required?

Comment (chewong) [reply config/jobs/kubernetes-sigs/sig-windows/sig-windows-config.yaml:186]: Are you referring to `--aksengine-agentpoolcount=2`?

**Pull request merged**